### PR TITLE
Improvement/refactor demo app

### PR DESCRIPTION
**Pull request by ethan605**
# Problem

As in ANSSI feature branch, we need to add a form to collect external testers' email and create checks

# Solution

Introduce new component in Demo app to collect email & create check after the flow completes. This PR is for refactoring the components in `src/demo` folder.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [n/a] Has the CHANGELOG been updated?
- [n/a] Has the README been updated?
- [n/a] Has the CONTRIBUTING doc been updated?
- [n/a] Has the RELEASE_GUIDELINES been updated?
- [n/a] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [n/a] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [n/a] Have any new automated tests been implemented or the existing ones changed?
- [n/a] Have any new manual tests been written down or the existing ones changed?
- [n/a] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1333-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1333-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1333-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-1333-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-1333-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://1333-pr-onfido-sdk-ui-onfido.surge.sh

File: src/demo/SidebarSections.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { h, FunctionComponent } from 'preact'
-import { useCallback, useRef } from 'preact/hooks'
+import { useCallback, useRef } from 'preact/compat'
```
Comment (stefaniacardenas) [inline src/demo/SidebarSections.tsx:2]: Why this change?

**Review by stefaniacardenas — APPROVED** [review]

File: src/demo/SidebarSections.tsx (hunk 1 above)
Comment (ethan605) [reply src/demo/SidebarSections.tsx:2]: Yeah just a refactor to align with hooks imports from other places. Preact said [there're 2 ways to import hooks](https://preactjs.com/guide/v10/hooks), and I think `preact/compat` is more generic here.
